### PR TITLE
Console safe

### PR DESCRIPTION
**Pull request by Lokathor**
* TODO: `read_console_output`

### CODE REVIEW

File: src/wincon.rs [hunk 1]
```diff
@@ -158,76 +175,85 @@ pub unsafe fn free_console() -> Result<(), ErrorCode> {
...
-    exe_w.as_mut_ptr(),
-  ) != 0
+  let needed_bytes =
+    unsafe { GetConsoleAliasesLengthW(exe_w.as_mut_ptr()) as usize };
+  // Note: technically this allocates more than we need, but whatever.
+  let mut buffer: Vec<u16> = vec![0; needed_bytes];
+  if unsafe {
+    GetConsoleAliasesW(
+      buffer.as_mut_ptr(),
+      buffer.capacity() as u32,
```
Comment (MaulingMonkey) [inline src/wincon.rs]: Might make this .len() out of paranoia.  This is only likely to matter if the vec overallocated and there's a race condition with another thread modifying aliases between this reading the length of the aliases and reading the actual buffer contents of the aliases....

File: src/wincon.rs [hunk 2]
```diff
@@ -780,19 +850,24 @@ pub unsafe fn write_console_output_attribute(
...
-    buffer.as_ptr(),
-    buffer.len().min(core::u32::MAX as usize) as u32,
-    position,
-    &mut num_written,
-  ) != 0
+  if unsafe {
+    WriteConsoleOutputCharacterW(
+      console_output,
+      buffer.as_ptr(),
+      buffer.len().min(core::u32::MAX as usize) as u32,
```
Comment (MaulingMonkey) [inline src/wincon.rs]: min now redundant

File: src/wincon.rs (hunk 1 above)
Comment (Lokathor) [reply src/wincon.rs]: done

File: src/wincon.rs (hunk 2 above)
Comment (Lokathor) [reply src/wincon.rs]: fixed

File: src/wincon.rs (hunk 1 above)
Comment (MaulingMonkey) [reply src/wincon.rs]: (if you wanted to be extra paranoid you might make get_console_aliases loop/retry a few times if GetLastError() == ERROR_INSUFFICIENT_BUFFER ...)

**Comment by Lokathor** [comment]
closed for likely blatent and uncontrollable unsafety

**Pull request closed without merging**